keygen/keygen_network.c

```diff
@@ -77,6 +77,9 @@ keygen_network_register(struct register_internal * C)
 	/* Success! */
 	return (0);
 
+err2:
+	netpacket_close(NPC);
+	network_fini();
```

Comment:
netpacket_close definitely belongs here, since NPC was allocated here; but network_fini should be called from the same function regardless of whether we're succeeding or failing.

That said, maybe the network_fini from keygen_actual.c belongs here?